smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/transactions/KafkaTransactions.java

```diff
@@ -0,0 +1,113 @@
+package io.smallrye.reactive.messaging.kafka.transactions;
+
+import java.util.function.Function;
+
+import org.eclipse.microprofile.reactive.messaging.Message;
+
+import io.smallrye.common.annotation.Experimental;
+import io.smallrye.mutiny.Uni;
+import io.smallrye.reactive.messaging.MutinyEmitter;
+
+/**
+ * Emitter interface for producing records in a Kafka transaction.
+ * This interface defines a custom Emitter for implementing Kafka Producer Transactions and Exactly-once processing.
+ * <p>
+ * Transactional producer:
+ *
+ * <pre>
+ * <code>
+ * &#64;Inject @Channel("tx-out-channel") KafkaTransactions&lt;String&gt; txEmitter;
+ *
+ * // ...
+ * Uni&lt;Integer&gt; txWork = txEmitter.withTransactions(emitter -> {
+ *     emitter.send("a");
+ *     emitter.send("b");
+ *     emitter.send("c");
+ *     return Uni.createFrom().item(3);
+ * });
+ *
+ * // ...
+ * </code>
+ * </pre>
+ *
+ * <p>
+ * Exactly-once processing:
+ *
+ * <pre>
+ * <code>
+ * &#64;Inject @Channel("tx-out-channel") KafkaTransactions&lt;String&gt; txEmitter;
+ *
+ * &#64;Incoming("in-channel")
+ * Uni&lt;Void&gt; process(KafkaRecordBatch&lt;String, String&gt; batch) {
+ * return txEmitter.withTransactions(batch, emitter -> {
+ * for (KafkaRecord&lt;String, String&gt; rec : batch) {
+ * txEmitter.send(KafkaRecord.of(record.getKey(), record.getPayload()));
+ * }
+ * return Uni.createFrom().voidItem();
+ * });
+ * }
+ *
+ *
```

Comment:
This javadoc (and perhaps other places, if I recall correctly?) says `withTransactions`, but the actual method is called `withTransaction`.